app/views/api/logs.py

```diff
@@ -0,0 +1,142 @@
+
+from . import CommonSchema, Id
+
+
+blp = Blueprint(Log.__tablename__ + 's', __name__)
+
+EVENT_DESC = (
+    'The event associated with this log entry',
+    'Filter by event'
+)
+USER_ID_DESC = (
+    'Source user for this log entry',
+    'Filter for user associated with this log entry'
+)
+ROOM_ID_DESC = (
+    'Source room for this log entry',
+    'Filter for room associated with this log entry'
+)
+RECEIVER_DESC = (
+    'Receiver associated  with this log entry',
+    'Filter for receiver associated with this log entry'
+)
+DATA_DESC = (
+    'Data stored inside this log entry',
+)
+
+
+# Only two schemas are needed but four are used to prettify OpenAPI Documentation
+class LogSchema(CommonSchema, SQLAlchemySchema):
+    class Meta:
+        model = Log
+
+    event = auto_field(metadata={'description': EVENT_DESC[0]})
+    user_id = Id(table=User, allow_none=True, metadata={'description': USER_ID_DESC[0]})
+    room_id = Id(table=Room, allow_none=True, metadata={'description': ROOM_ID_DESC[0]})
+    receiver = Id(table=User, allow_none=True, metadata={'description': RECEIVER_DESC[0]})
+    data = ma.fields.Dict(required=True, metadata={'description': DATA_DESC[0]})
+
+
+class LogResponseSchema(LogSchema):
+    event = auto_field(required=False, metadata={'description': EVENT_DESC[0]})
+    data = ma.fields.Dict(metadata={'description': DATA_DESC[0]})
+
+
+class LogUpdateSchema(LogResponseSchema):
+    event = auto_field(required=False, allow_none=True, metadata={'description': EVENT_DESC[0]})
+    user_id = Id(table=User, allow_none=True, metadata={'description': USER_ID_DESC[0]})
+    room_id = Id(table=Room, allow_none=True, metadata={'description': ROOM_ID_DESC[0]})
+    receiver = Id(table=User, allow_none=True, metadata={'description': RECEIVER_DESC[0]})
+    data = ma.fields.Dict(required=False, allow_none=True, metadata={'description': DATA_DESC[0]})
+
+
+class LogQuerySchema(LogSchema):
+    event = auto_field(required=False, allow_none=True, metadata={'description': EVENT_DESC[1]})
+    user_id = Id(table=User, allow_none=True, metadata={'description': USER_ID_DESC[1]})
+    room_id = Id(table=Room, allow_none=True, metadata={'description': ROOM_ID_DESC[1]})
+    receiver = Id(table=User, allow_none=True, metadata={'description': RECEIVER_DESC[1]})
+    data = auto_field(dump_only=True)
+
+
+@blp.route('/')
+class Logs(MethodView):
+    @blp.etag
+    @blp.arguments(LogQuerySchema, location='query')
+    @blp.response(200, LogResponseSchema(many=True))
+    @blp.paginate()
+    def get(self, args, pagination_parameters):
+        """List logs"""
+        db = current_app.session
+        query = db.query(Log) \
+            .filter_by(**args) \
+            .order_by(Log.date_created.desc())
+        pagination_parameters.item_count = query.count()
+        return query \
+            .limit(pagination_parameters.page_size) \
+            .offset(pagination_parameters.first_item) \
+            .all()
```

Comment:
```suggestion
    def get(self, args):
        """List logs"""
        db = current_app.session
        return db.query(Log) \
            .filter_by(**args) \
            .order_by(Log.date_created.desc()) \
            .all()
```

Also applies to other `@blp.paginate` decorators in other views.